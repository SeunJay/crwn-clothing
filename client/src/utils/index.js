import { applyMiddleware, createStore } from "redux";
import rootReducer from "../redux/root-reducer";
import { middlewares } from "../redux/store";

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );
  return createStoreWithMiddleware(rootReducer, initialState);
};
