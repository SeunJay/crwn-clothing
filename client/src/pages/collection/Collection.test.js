import React from "react";
import { shallow } from "enzyme";
import { testStore } from "../../utils";
import Collection from "./Collection";

const setUp = (initialState = {}) => {

  const store = testStore(initialState);
  const wrapper = shallow(<Collection store={store} />);
  return wrapper;
};

describe("Collection Page", () => {
   let wrapper;

    beforeEach(() => {
      const initialState = {
        shop: {
          mockItems: [{ id: 1 }, { id: 2 }, { id: 3 }]
        },
      };

      wrapper = setUp(initialState);
    });


      it("should render the CollectionPage component", () => {
        expect(wrapper).toMatchSnapshot();
      });
});
