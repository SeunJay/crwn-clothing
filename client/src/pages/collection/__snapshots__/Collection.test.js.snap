// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Page should render the CollectionPage component 1`] = `undefined`;
